W-14659026-SFDC-Bridge-se (#359)
mulesoft/docs-rpa: rpa-home/modules/ROOT/pages/invoke-rpa-from-flow-run.adoc

# Use a published RPA Process in Salesforce Flow

Publishing an invocable run configuration in RPA Manager, creates an External MuleSoft Service in Salesforce. The service's name consists of the name of the RPA Manager automation project without the blanks, followed by a '10'. 

> Follow these steps to use your process in a Salesforce Flow:

Reviewer: I would swap this wording:
"To use your process..."

. Create a new flow. In this example we use a screen flow for illustrational purposes.
. Create a resource for the process execution input:
+ 

> image:invocableflow-resource-processExecution.png[Resource for the Process Execution]

Reviewer: Required now

```suggestion
image:invocableflow-resource-processExecution.png[Resource for the process execution]
```

+
[cols="1,2"]
|===
|*Field*
|*Value*

|Resource Type

> |Select *Variable*.

Reviewer: The initial verbs are not needed. I marked the first two entries.

> |Select *Variable*.

Reviewer: ```suggestion
|*Variable*.
```

> |Select *Variable*.

Reviewer: ```suggestion
|*Variable*.
```

|API Name

> |Choose a name for this resource; for example `ProcessParameters`.

Reviewer: ```suggestion
|Name of this resource; for example `ProcessParameters`.
```

|Data Type
|Select *Apex-Defined*.

|Apex Class
|Select the Apex class *ExternalService__<RPA Project Name>10_ProcessExecution*, for example `ExternalService__ExampleProcess10_ProcessExecution``, 
|===

. Use a screen to set the input parameters for debugging purposes:
+

> image:invocableflow-screen-inputvariables.png[]

Reviewer: Required now - add all missing alt text 

Missing alt text

. Assign all required values to the process parameters:
+ 

> image:invocableflow-assignment-processExecution.png[]

Reviewer: Missing alt text

+
The variable names are the same as the Activity Parameter names defined in RPA Builder:
+

> image:invocableflow-activityparameters.png[]

Reviewer: I will stop marking missing alt tex going forward.

. Choose the *Start Process* Action. You can filter for your process by name.
+
[cols="1,2"]
|===
|*Process Start Type*
|*Application*

|*Start Process V2*
|Ensures that only one instance of the process is running at the same time.

|*Start Process Non Idempotent V2*
|Enables parallel execution of the same process.

|===
+
image:invocableflow-action-choosestarttype.png[]

> . Configure the action filling the body with the resource you defined:

Reviewer: Required now

```suggestion
. Configure the action by filling the body with the resource you defined:
```

Reviewer: ...that you defined earlier ???

+
image:invocableflow-action-startconfiguration.png[]

> . Get the execution Status for the first time:

Reviewer: Required now

```suggestion
. Get the execution status for the first time:
```

+
image:invocableflow-action-getfirstexecutionstatus.png[]

> . Create a resource for the status of the process execution:

Reviewer: "for the process execution" > "for the execution status"

+
image:invocableflow-resource-processExecutionStatus.png[]

> . Create a resource for collecting all states the started process returns while running:

Reviewer: ```suggestion
. Create a resource for collecting all states that the started process returns while it runs:
```

+
image:invocableflow-resource-processExecutionStatusCollection.png[]

. Add the first returned status to the status collection:
+
image:invocableflow-assignment-processExecutionStatus1.png[]

> . Since the run time of a process is not known, add a loop over all returned status:

Reviewer: Required now

```suggestion
. Since the run time of a process is not known, add a loop over all returned statuses:
```

+
image:invocableflow-loop.png[]

. To check for the current status, define three text constants: `ìsStarted`, `running`, `error`.
. Add a decision element:
+
image:invocableflow-decision-checksucces.png[]

> . If the returned status is not success, add the current status to the loop collection:

Reviewer: ```suggestion
. If the returned status is not `Success`, add the current status to the loop collection:
```

Reviewer: In the edit above, I assume that `Success` is returned instead of a code.

+
image:invocableflow-assignment-processExecutionStatusN.png[]
+

> This is repeated until the proces has finished and the status is success.

Reviewer: ```suggestion
This repeats until the process finishes 
and the status is `Success`.
```

Reviewer: What happens if there is an error? Does the process keep looping or does it end?

Author: I don't know. I guess there has to be some kind of error handling. We have to check this with Salesforce developers.
